// ng-zorro-antd/esm5/core/overlay/nz-overlay.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: nz-overlay.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { NgModule } from '@angular/core';
import { NzConnectedOverlayDirective } from './nz-connected-overlay';
import * as ɵngcc0 from '@angular/core';
var NzOverlayModule = /** @class */ (function () {
    function NzOverlayModule() {
    }
NzOverlayModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NzOverlayModule });
NzOverlayModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NzOverlayModule_Factory(t) { return new (t || NzOverlayModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NzOverlayModule, { declarations: function () { return [NzConnectedOverlayDirective]; }, exports: function () { return [NzConnectedOverlayDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NzOverlayModule, [{
        type: NgModule,
        args: [{
                declarations: [NzConnectedOverlayDirective],
                exports: [NzConnectedOverlayDirective]
            }]
    }], function () { return []; }, null); })();
    return NzOverlayModule;
}());
export { NzOverlayModule };
