// ngx-filesize/esm2015/lib/ngx-filesize.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/ngx-filesize.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { FileSizePipe } from './filesize.pipe';
import * as ɵngcc0 from '@angular/core';
export class NgxFilesizeModule {
}
NgxFilesizeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxFilesizeModule });
NgxFilesizeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxFilesizeModule_Factory(t) { return new (t || NgxFilesizeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxFilesizeModule, { declarations: function () { return [FileSizePipe]; }, exports: function () { return [FileSizePipe]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxFilesizeModule, [{
        type: NgModule,
        args: [{
                declarations: [FileSizePipe],
                exports: [FileSizePipe]
            }]
    }], null, null); })();
