// ng-chartjs/esm5/lib/ng-chartjs.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Injectable, Inject, forwardRef } from '@angular/core';
import { Chart } from 'chart.js';
import { NgChartjsCustomPluginToken } from './plugin-token';
import { PluginConfig } from './pluginsConfig';
import { StoreService } from './store.service';
import * as i0 from "@angular/core";
import * as i1 from "./plugin-token";
import * as i2 from "./store.service";
import * as ɵngcc0 from '@angular/core';
var NgChartjsService = /** @class */ (function () {
    function NgChartjsService(pluginConfig, storeService) {
        this.pluginConfig = pluginConfig;
        this.storeService = storeService;
        this.plugins = pluginConfig.plugins;
        if (this.plugins.length !== 0 || this.plugins) {
            for (var i = 0; i < this.plugins.length; i++) {
                if (this.plugins[i]) {
                    Chart.plugins.register(this.plugins[i]);
                }
            }
        }
    }
    // get chart instance by id
    /**
     * @param {?} id
     * @return {?}
     */
    NgChartjsService.prototype.getChart = /**
     * @param {?} id
     * @return {?}
     */
    function (id) {
        return this.storeService.getChart(id);
    };
    /** @nocollapse */
    NgChartjsService.ctorParameters = function () { return [
        { type: PluginConfig, decorators: [{ type: Inject, args: [forwardRef(function () { return NgChartjsCustomPluginToken; }),] }] },
        { type: StoreService }
    ]; };
    /** @nocollapse */ NgChartjsService.ngInjectableDef = i0.defineInjectable({ factory: function NgChartjsService_Factory() { return new NgChartjsService(i0.inject(i1.NgChartjsCustomPluginToken), i0.inject(i2.StoreService)); }, token: NgChartjsService, providedIn: "root" });
NgChartjsService.ɵfac = function NgChartjsService_Factory(t) { return new (t || NgChartjsService)(ɵngcc0.ɵɵinject(forwardRef(function () { return NgChartjsCustomPluginToken; })), ɵngcc0.ɵɵinject(StoreService)); };
NgChartjsService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: NgChartjsService, factory: function (t) { return NgChartjsService.ɵfac(t); }, providedIn: 'root' });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgChartjsService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: PluginConfig, decorators: [{
                type: Inject,
                args: [forwardRef(function () { return NgChartjsCustomPluginToken; })]
            }] }, { type: StoreService }]; }, null); })();
    return NgChartjsService;
}());
export { NgChartjsService };
if (false) {
    /** @type {?} */
    NgChartjsService.prototype.plugins;
    /** @type {?} */
    NgChartjsService.prototype.pluginConfig;
    /** @type {?} */
    NgChartjsService.prototype.storeService;
}
