// ngx-filesize/bundles/ngx-filesize.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),exports, require('@angular/core'), require('filesize')) :
    typeof define === 'function' && define.amd ? define('ngx-filesize', ['@angular/core','exports', '@angular/core', 'filesize'], factory) :
    (global = global || self, factory(global.ng.core,global['ngx-filesize'] = {}, global.ng.core, global.filesize_));
}(this, (function (ɵngcc0,exports, core, filesize_) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/filesize.pipe.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var FileSizePipe = /** @class */ (function () {
        function FileSizePipe() {
        }
        /**
         * @private
         * @param {?} value
         * @param {?=} options
         * @return {?}
         */
        FileSizePipe.transformOne = function (value, options) {
            /** @type {?} */
            var filesize = filesize_;
            return filesize(value, options);
        };
        /**
         * @param {?} value
         * @param {?=} options
         * @return {?}
         */
        FileSizePipe.prototype.transform = function (value, options) {
            if (Array.isArray(value)) {
                return value.map(( /**
                 * @param {?} val
                 * @return {?}
                 */function (/**
                 * @param {?} val
                 * @return {?}
                 */ val) { return FileSizePipe.transformOne(val, options); }));
            }
            return FileSizePipe.transformOne(value, options);
        };
FileSizePipe.ɵfac = function FileSizePipe_Factory(t) { return new (t || FileSizePipe)(); };
FileSizePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "filesize", type: FileSizePipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FileSizePipe, [{
        type: core.Pipe,
        args: [{
                name: 'filesize'
            }]
    }], function () { return []; }, null); })();
        return FileSizePipe;
    }());

    /**
     * @fileoverview added by tsickle
     * Generated from: lib/ngx-filesize.module.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgxFilesizeModule = /** @class */ (function () {
        function NgxFilesizeModule() {
        }
NgxFilesizeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxFilesizeModule });
NgxFilesizeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxFilesizeModule_Factory(t) { return new (t || NgxFilesizeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxFilesizeModule, { declarations: [FileSizePipe], exports: [FileSizePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxFilesizeModule, [{
        type: core.NgModule,
        args: [{
                declarations: [FileSizePipe],
                exports: [FileSizePipe]
            }]
    }], function () { return []; }, null); })();
        return NgxFilesizeModule;
    }());

    /**
     * @fileoverview added by tsickle
     * Generated from: public-api.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * Generated from: ngx-filesize.ts
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    exports.FileSizePipe = FileSizePipe;
    exports.NgxFilesizeModule = NgxFilesizeModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
