// ng-zorro-antd/esm5/ng-zorro-antd.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: ng-zorro-antd.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { NgModule } from '@angular/core';
import { NzAffixModule } from 'ng-zorro-antd/affix';
import { NzAlertModule } from 'ng-zorro-antd/alert';
import { NzAnchorModule } from 'ng-zorro-antd/anchor';
import { NzAutocompleteModule } from 'ng-zorro-antd/auto-complete';
import { NzAvatarModule } from 'ng-zorro-antd/avatar';
import { NzBackTopModule } from 'ng-zorro-antd/back-top';
import { NzBadgeModule } from 'ng-zorro-antd/badge';
import { NzBreadCrumbModule } from 'ng-zorro-antd/breadcrumb';
import { NzButtonModule } from 'ng-zorro-antd/button';
import { NzCalendarModule } from 'ng-zorro-antd/calendar';
import { NzCardModule } from 'ng-zorro-antd/card';
import { NzCarouselModule } from 'ng-zorro-antd/carousel';
import { NzCascaderModule } from 'ng-zorro-antd/cascader';
import { NzCheckboxModule } from 'ng-zorro-antd/checkbox';
import { NzCollapseModule } from 'ng-zorro-antd/collapse';
import { NzCommentModule } from 'ng-zorro-antd/comment';
import { warnDeprecation } from 'ng-zorro-antd/core/logger';
import { NzNoAnimationModule } from 'ng-zorro-antd/core/no-animation';
import { NzTransButtonModule } from 'ng-zorro-antd/core/trans-button';
import { NzWaveModule } from 'ng-zorro-antd/core/wave';
import { NzDatePickerModule } from 'ng-zorro-antd/date-picker';
import { NzDescriptionsModule } from 'ng-zorro-antd/descriptions';
import { NzDividerModule } from 'ng-zorro-antd/divider';
import { NzDrawerModule } from 'ng-zorro-antd/drawer';
import { NzDropDownModule } from 'ng-zorro-antd/dropdown';
import { NzEmptyModule } from 'ng-zorro-antd/empty';
import { NzFormModule } from 'ng-zorro-antd/form';
import { NzGridModule } from 'ng-zorro-antd/grid';
import { NzI18nModule } from 'ng-zorro-antd/i18n';
import { NzIconModule } from 'ng-zorro-antd/icon';
import { NzInputModule } from 'ng-zorro-antd/input';
import { NzInputNumberModule } from 'ng-zorro-antd/input-number';
import { NzLayoutModule } from 'ng-zorro-antd/layout';
import { NzListModule } from 'ng-zorro-antd/list';
import { NzMentionModule } from 'ng-zorro-antd/mention';
import { NzMenuModule } from 'ng-zorro-antd/menu';
import { NzMessageModule } from 'ng-zorro-antd/message';
import { NzModalModule } from 'ng-zorro-antd/modal';
import { NzNotificationModule } from 'ng-zorro-antd/notification';
import { NzPageHeaderModule } from 'ng-zorro-antd/page-header';
import { NzPaginationModule } from 'ng-zorro-antd/pagination';
import { NzPopconfirmModule } from 'ng-zorro-antd/popconfirm';
import { NzPopoverModule } from 'ng-zorro-antd/popover';
import { NzProgressModule } from 'ng-zorro-antd/progress';
import { NzRadioModule } from 'ng-zorro-antd/radio';
import { NzRateModule } from 'ng-zorro-antd/rate';
import { NzResultModule } from 'ng-zorro-antd/result';
import { NzSelectModule } from 'ng-zorro-antd/select';
import { NzSkeletonModule } from 'ng-zorro-antd/skeleton';
import { NzSliderModule } from 'ng-zorro-antd/slider';
import { NzSpinModule } from 'ng-zorro-antd/spin';
import { NzStatisticModule } from 'ng-zorro-antd/statistic';
import { NzStepsModule } from 'ng-zorro-antd/steps';
import { NzSwitchModule } from 'ng-zorro-antd/switch';
import { NzTableModule } from 'ng-zorro-antd/table';
import { NzTabsModule } from 'ng-zorro-antd/tabs';
import { NzTagModule } from 'ng-zorro-antd/tag';
import { NzTimePickerModule } from 'ng-zorro-antd/time-picker';
import { NzTimelineModule } from 'ng-zorro-antd/timeline';
import { NzToolTipModule } from 'ng-zorro-antd/tooltip';
import { NzTransferModule } from 'ng-zorro-antd/transfer';
import { NzTreeModule } from 'ng-zorro-antd/tree';
import { NzTreeSelectModule } from 'ng-zorro-antd/tree-select';
import { NzTypographyModule } from 'ng-zorro-antd/typography';
import { NzUploadModule } from 'ng-zorro-antd/upload';
import * as ɵngcc0 from '@angular/core';
export { NzAffixComponent, NzAffixModule } from 'ng-zorro-antd/affix';
export { NzAlertComponent, NzAlertModule } from 'ng-zorro-antd/alert';
export { NzAnchorLinkComponent, NzAnchorComponent, NzAnchorModule } from 'ng-zorro-antd/anchor';
export { NzAutocompleteModule, NzAutocompleteComponent, getNzAutocompleteMissingPanelError, NZ_AUTOCOMPLETE_VALUE_ACCESSOR, NzAutocompleteTriggerDirective, NzOptionSelectionChange, NzAutocompleteOptionComponent, NzAutocompleteOptgroupComponent } from 'ng-zorro-antd/auto-complete';
export { NzAvatarComponent, NzAvatarModule } from 'ng-zorro-antd/avatar';
export { NzBackTopComponent, NzBackTopModule } from 'ng-zorro-antd/back-top';
export { NzBadgeComponent, NzBadgeModule } from 'ng-zorro-antd/badge';
export { NzBreadCrumbItemComponent, NzBreadCrumbComponent, NzBreadCrumbModule, NzBreadCrumbSeparatorComponent } from 'ng-zorro-antd/breadcrumb';
export { NzButtonComponent, NzButtonGroupComponent, NzButtonModule } from 'ng-zorro-antd/button';
export { NzCalendarModule, NzCalendarComponent, NzDateCellDirective, NzMonthCellDirective, NzDateFullCellDirective, NzMonthFullCellDirective, NzCalendarHeaderComponent } from 'ng-zorro-antd/calendar';
export { NzCardGridDirective, NzCardComponent, NzCardModule, NzCardLoadingComponent, NzCardMetaComponent, NzCardTabComponent } from 'ng-zorro-antd/card';
export { NzCarouselModule, NzCarouselComponent, NzCarouselContentDirective, NzCarouselBaseStrategy, NZ_CAROUSEL_CUSTOM_STRATEGIES } from 'ng-zorro-antd/carousel';
export { isShowSearchObject, isChildOption, isParentOption, NzCascaderComponent, NzCascaderModule, NzCascaderService, NzCascaderOptionComponent } from 'ng-zorro-antd/cascader';
export { NzCheckboxComponent, NzCheckboxModule, NzCheckboxGroupComponent, NzCheckboxWrapperComponent } from 'ng-zorro-antd/checkbox';
export { NzCollapsePanelComponent, NzCollapseComponent, NzCollapseModule } from 'ng-zorro-antd/collapse';
export { NzCommentModule, NzCommentComponent, NzCommentAvatarDirective, NzCommentContentDirective, NzCommentActionHostDirective, NzCommentActionComponent } from 'ng-zorro-antd/comment';
export { NzDatePickerModule, NzDatePickerComponent, NzRangePickerComponent, NzMonthPickerComponent, NzWeekPickerComponent, NzYearPickerComponent, ɵDatePickerService, ɵDateRangePopupComponent, ɵInnerPopupComponent, ɵNzPickerComponent, ɵCalendarFooterComponent, getTimeConfig, isTimeValidByConfig, isTimeValid, isAllowedDate, PREFIX_CLASS, ɵDecadeHeaderComponent, ɵDecadeTableComponent, ɵYearHeaderComponent, ɵYearTableComponent, ɵMonthHeaderComponent, ɵMonthTableComponent, ɵDateHeaderComponent, ɵDateTableComponent, ɵAbstractPanelHeader, ɵAbstractTable, LibPackerModule, transCompatFormat } from 'ng-zorro-antd/date-picker';
export { NzDescriptionsModule, NzDescriptionsComponent, NzDescriptionsItemComponent } from 'ng-zorro-antd/descriptions';
export { NzDividerComponent, NzDividerModule } from 'ng-zorro-antd/divider';
export { DRAWER_ANIMATE_DURATION, NzDrawerComponent, NzDrawerModule, DrawerBuilderForService, NzDrawerService, NzDrawerServiceModule, NzDrawerRef } from 'ng-zorro-antd/drawer';
export { NzDropDownDirective, NzDropDownModule, NzDropDownADirective, NzDropdownButtonDirective, NzDropdownMenuComponent, NzContextMenuService, NzContextMenuServiceModule } from 'ng-zorro-antd/dropdown';
export { NzEmptySimpleComponent, NzEmptyDefaultComponent, NzEmbedEmptyComponent, NzEmptyComponent, NzEmptyModule, NZ_EMPTY_COMPONENT_NAME } from 'ng-zorro-antd/empty';
export { NzFormModule, NzFormDirective, NzFormItemComponent, NzFormLabelComponent, NzFormControlComponent, NzFormTextComponent, NzFormSplitComponent } from 'ng-zorro-antd/form';
export { NzRowDirective, NzColDirective, NzGridModule } from 'ng-zorro-antd/grid';
export { ar_EG, az_AZ, bg_BG, ca_ES, cs_CZ, da_DK, de_DE, el_GR, en_GB, en_US, es_ES, et_EE, fa_IR, fi_FI, fr_BE, fr_FR, he_IL, hi_IN, hr_HR, hu_HU, hy_AM, id_ID, is_IS, it_IT, ja_JP, ka_GE, kn_IN, ko_KR, ku_IQ, lv_LV, mk_MK, mn_MN, ms_MY, nb_NO, ne_NP, nl_BE, nl_NL, pl_PL, pt_BR, pt_PT, ro_RO, ru_RU, sk_SK, sl_SI, sr_RS, sv_SE, ta_IN, th_TH, tr_TR, uk_UA, vi_VN, zh_CN, zh_TW, NzI18nModule, NzI18nService, mergeDateConfig, NZ_DATE_CONFIG, NZ_DATE_CONFIG_DEFAULT, NZ_DATE_FNS_COMPATIBLE, NZ_I18N, NZ_DATE_LOCALE, NzI18nPipe, DATE_HELPER_SERVICE_FACTORY, DateHelperService, DateHelperByDateFns, DateHelperByDatePipe, convertTokens } from 'ng-zorro-antd/i18n';
export { NzIconModule, NzIconDirective, NZ_ICONS, NZ_ICON_DEFAULT_TWOTONE_COLOR, DEFAULT_TWOTONE_COLOR, NzIconService, NZ_ICONS_PATCH, NzIconPatchService } from 'ng-zorro-antd/icon';
export { NzInputGroupWhitSuffixOrPrefixDirective, NzInputGroupComponent, NzInputModule, NzInputGroupSlotComponent, NzInputDirective, NzAutosizeDirective } from 'ng-zorro-antd/input';
export { NzInputNumberComponent, NzInputNumberModule } from 'ng-zorro-antd/input-number';
export { ɵNzSiderTriggerComponent, NzContentComponent, NzFooterComponent, NzHeaderComponent, NzLayoutComponent, NzSiderComponent, NzLayoutModule } from 'ng-zorro-antd/layout';
export { NzListItemMetaComponent, NzListItemComponent, NzListComponent, NzListModule, NzListEmptyComponent, NzListHeaderComponent, NzListFooterComponent, NzListPaginationComponent, NzListLoadMoreDirective, NzListGridDirective, NzListItemExtraComponent, NzListItemActionComponent, NzListItemActionsComponent, NzListItemMetaTitleComponent, NzListItemMetaDescriptionComponent, NzListItemMetaAvatarComponent } from 'ng-zorro-antd/list';
export { NzMentionModule, NzMentionComponent, NZ_MENTION_TRIGGER_ACCESSOR, NzMentionTriggerDirective, NzMentionSuggestionDirective, NzMentionService } from 'ng-zorro-antd/mention';
export { MenuServiceFactory, MenuDropDownTokenFactory, NzMenuDirective, MenuGroupFactory, NzMenuGroupComponent, NzMenuDividerDirective, NzMenuItemDirective, NzSubMenuComponent, NzSubMenuTitleComponent, NzSubmenuInlineChildComponent, NzSubmenuNoneInlineChildComponent, NzMenuModule, NzSubmenuService, MenuService, NzIsMenuInsideDropDownToken, NzMenuServiceLocalToken } from 'ng-zorro-antd/menu';
export { NzMNService, NzMNContainerComponent, NzMNComponent, NzMessageService, NzMessageServiceModule, NzMessageModule, NzMessageComponent, NzMessageContainerComponent } from 'ng-zorro-antd/message';
export { ModalOptions, NzModalService, NzModalRef, ZOOM_CLASS_NAME_MAP, FADE_CLASS_NAME_MAP, MODAL_MASK_CLASS_NAME, NZ_CONFIG_COMPONENT_NAME, NzModalComponent, NzModalFooterDirective, NzModalModule, NzModalConfirmContainerComponent, NzModalContainerComponent, throwNzModalContentAlreadyAttachedError, BaseModalContainer, NzModalCloseComponent, NzModalTitleComponent, NzModalLegacyAPI, nzModalAnimations, NzModalFooterComponent, applyConfigDefaults, getValueWithConfig, setContentInstanceParams, getConfigFromComponent } from 'ng-zorro-antd/modal';
export { NzNotificationComponent, NzNotificationModule, NzNotificationService, NzNotificationServiceModule, NzNotificationContainerComponent } from 'ng-zorro-antd/notification';
export { NzPageHeaderModule, NzPageHeaderComponent, NzPageHeaderTitleDirective, NzPageHeaderSubtitleDirective, NzPageHeaderContentDirective, NzPageHeaderTagDirective, NzPageHeaderExtraDirective, NzPageHeaderFooterDirective, NzPageHeaderBreadcrumbDirective, NzPageHeaderAvatarDirective } from 'ng-zorro-antd/page-header';
export { NzPaginationComponent, NzPaginationModule, NzPaginationSimpleComponent, NzPaginationOptionsComponent, NzPaginationItemComponent, NzPaginationDefaultComponent } from 'ng-zorro-antd/pagination';
export { NzPopconfirmDirective, NzPopconfirmComponent, NzPopconfirmModule } from 'ng-zorro-antd/popconfirm';
export { NzPopoverDirective, NzPopoverComponent, NzPopoverModule } from 'ng-zorro-antd/popover';
export { NzProgressModule, NzProgressComponent } from 'ng-zorro-antd/progress';
export { NzRadioButtonDirective, NzRadioGroupComponent, NzRadioComponent, NzRadioService, NzRadioModule } from 'ng-zorro-antd/radio';
export { NzRateComponent, NzRateModule, NzRateItemComponent } from 'ng-zorro-antd/rate';
export { ɵNzResultNotFoundComponent, ɵNzResultServerErrorComponent, ɵNzResultUnauthorizedComponent, NzResultModule, NzResultComponent, NzResultTitleDirective, NzResultSubtitleDirective, NzResultIconDirective, NzResultContentDirective, NzResultExtraDirective } from 'ng-zorro-antd/result';
export { NzOptionGroupComponent, NzOptionContainerComponent, NzOptionComponent, NzSelectComponent, NzSelectModule, NzOptionItemComponent, NzOptionItemGroupComponent, NzSelectTopControlComponent, NzSelectSearchComponent, NzSelectItemComponent, NzSelectClearComponent, NzSelectArrowComponent, NzSelectPlaceholderComponent } from 'ng-zorro-antd/select';
export { NzSkeletonComponent, NzSkeletonModule, NzSkeletonElementDirective, NzSkeletonElementButtonComponent, NzSkeletonElementAvatarComponent, NzSkeletonElementInputComponent } from 'ng-zorro-antd/skeleton';
export { NzSliderComponent, ɵNzSliderService, NzSliderModule, ɵNzSliderHandleComponent, ɵNzSliderMarksComponent, ɵNzSliderStepComponent, ɵNzSliderTrackComponent, NzMarks } from 'ng-zorro-antd/slider';
export { NzSpinComponent, NzSpinModule } from 'ng-zorro-antd/spin';
export { NzCountdownComponent, NzStatisticComponent, NzStatisticModule, NzStatisticNumberComponent } from 'ng-zorro-antd/statistic';
export { NzStepsComponent, NzStepComponent, NzStepsModule } from 'ng-zorro-antd/steps';
export { NzSwitchComponent, NzSwitchModule } from 'ng-zorro-antd/switch';
export { NzTableModule, NzTableDataService, NzTableStyleService, NzTableComponent, NzTableContentComponent, NzTableFixedRowComponent, NzTableInnerDefaultComponent, NzTableInnerScrollComponent, NzTableVirtualScrollDirective, NzTbodyComponent, NzTheadComponent, NzTrDirective, NzTrExpandDirective, NzTableTitleFooterComponent, NzTrMeasureComponent, NzCellFixedDirective, NzTableCellDirective, NzThMeasureDirective, NzTdAddOnComponent, NzThSelectionComponent, NzThAddOnComponent, NzTableFilterComponent, NzRowExpandButtonDirective, NzRowIndentDirective, NzTableSelectionComponent, NzTableSortersComponent, NzFilterTriggerComponent, NzCellAlignDirective, NzCellEllipsisDirective, NzCellBreakWordDirective } from 'ng-zorro-antd/table';
export { NzTabBodyComponent, NzTabLabelDirective, NzTabComponent, NzTabsInkBarDirective, NzTabsModule, NzTabsNavComponent, NzTabSetComponent, NzTabDirective, NzTabLinkDirective, NzTabChangeEvent } from 'ng-zorro-antd/tabs';
export { NzTagComponent, NzTagModule } from 'ng-zorro-antd/tag';
export { NzTimePickerComponent, NzTimePickerModule, NzTimePickerPanelComponent, NzTimeValueAccessorDirective } from 'ng-zorro-antd/time-picker';
export { NzTimelineItemComponent, NzTimelineComponent, NzTimelineModule, TimelineService } from 'ng-zorro-antd/timeline';
export { NzTooltipDirective, NzToolTipComponent, NzToolTipModule, isTooltipEmpty, NzTooltipBaseDirective, NzTooltipBaseComponent } from 'ng-zorro-antd/tooltip';
export { NzTransferListComponent, NzTransferSearchComponent, NzTransferComponent, NzTransferModule } from 'ng-zorro-antd/transfer';
export { NzTreeNode, NzTreeModule, NzTreeServiceFactory, NzTreeComponent, NzTreeNodeComponent, NzTreeIndentComponent, NzTreeService, NzTreeNodeSwitcherComponent, NzTreeNodeCheckboxComponent, NzTreeNodeTitleComponent } from 'ng-zorro-antd/tree';
export { higherOrderServiceFactory, NzTreeSelectComponent, NzTreeSelectModule, NzTreeSelectService } from 'ng-zorro-antd/tree-select';
export { NzTypographyModule, NzTypographyComponent, NzTextCopyComponent, NzTextEditComponent } from 'ng-zorro-antd/typography';
export { NzShowUploadListInterface, NzUploadBtnComponent, NzUploadListComponent, NzUploadComponent, NzUploadModule } from 'ng-zorro-antd/upload';
export { VERSION } from 'ng-zorro-antd/version';
export { AnimationDuration, AnimationCurves, collapseMotion, treeCollapseMotion, fadeMotion, helpMotion, moveUpMotion, notificationMotion, slideMotion, slideAlertMotion, zoomMotion, zoomBigMotion, zoomBadgeMotion } from 'ng-zorro-antd/core/animation';
export { WithConfig, NzConfigService, NZ_CONFIG } from 'ng-zorro-antd/core/config';
export { environment } from 'ng-zorro-antd/core/environments';
export { NzHighlightPipe, NzHighlightModule } from 'ng-zorro-antd/core/highlight';
export { PREFIX, warn, warnDeprecation, log } from 'ng-zorro-antd/core/logger';
export { NzNoAnimationModule, NzNoAnimationDirective } from 'ng-zorro-antd/core/no-animation';
export { NzOutletModule, NzStringTemplateOutletDirective } from 'ng-zorro-antd/core/outlet';
export { NzConnectedOverlayDirective, NzOverlayModule, getPlacementName, POSITION_MAP, DEFAULT_TOOLTIP_POSITIONS, DEFAULT_CASCADER_POSITIONS, DEFAULT_MENTION_TOP_POSITIONS, DEFAULT_MENTION_BOTTOM_POSITIONS } from 'ng-zorro-antd/core/overlay';
export { NzPipesModule, NzTimeRangePipe, NzToCssUnitPipe } from 'ng-zorro-antd/core/pipe';
export { cancelRequestAnimationFrame, reqAnimFrame } from 'ng-zorro-antd/core/polyfill';
export { ɵa, NzResizeObserversModule, NzResizeObserver } from 'ng-zorro-antd/core/resize-observers';
export { NzResizeService, NzSingletonService, NzDragService, NzScrollService, NzBreakpointEnum, gridResponsiveMap, siderResponsiveMap, NzBreakpointService } from 'ng-zorro-antd/core/services';
export { ɵcreateComponentBed, dispatchEvent, dispatchFakeEvent, dispatchKeyboardEvent, dispatchMouseEvent, dispatchTouchEvent, createMouseEvent, createTouchEvent, createKeyboardEvent, createFakeEvent, typeInElement, wrappedErrorMessage, FakeViewportRuler, MockNgZone } from 'ng-zorro-antd/core/testing';
export { sortRangeValue, normalizeRangeValue, cloneDate, CandyDate, timeUnits } from 'ng-zorro-antd/core/time';
export { NzTransButtonModule, NzTransButtonDirective } from 'ng-zorro-antd/core/trans-button';
export { ɵNzTransitionPatchModule, ɵNzTransitionPatchDirective } from 'ng-zorro-antd/core/transition-patch';
export { NzTreeBaseService, NzTreeHigherOrderServiceToken, NzTreeBase, isCheckDisabled, isInArray, getPosition, getKey, flattenTreeData } from 'ng-zorro-antd/core/tree';
export {} from 'ng-zorro-antd/core/types';
export { toArray, arraysEqual, shallowCopyArray, isNotNil, isNil, shallowEqual, isInteger, isEmpty, filterNotEmptyNode, isNonEmptyString, isTemplateRef, isComponent, toBoolean, toNumber, toCssPixel, valueFunctionProp, InputBoolean, InputCssPixel, InputNumber, silentEvent, getElementOffset, isTouchEvent, getEventPosition, getRegExp, getMentions, padStart, padEnd, getRepeatedElement, isPromise, getPercent, getPrecision, ensureNumberInRange, scrollIntoView, getCaretCoordinates, createDebugEle, properties, isStyleSupport, getStyleAsText, pxToNumber, measure, measureScrollbar, ensureInBounds, inNextTick, wrapIntoObservable } from 'ng-zorro-antd/core/util';
export { NzWaveRenderer, NZ_WAVE_GLOBAL_CONFIG_FACTORY, NZ_WAVE_GLOBAL_DEFAULT_CONFIG, NZ_WAVE_GLOBAL_CONFIG, NzWaveDirective, NzWaveModule } from 'ng-zorro-antd/core/wave';
var NgZorroAntdModule = /** @class */ (function () {
    function NgZorroAntdModule() {
        warnDeprecation("The `NgZorroAntdModule` has been deprecated and will be removed in 10.0.0. Please use secondary entry instead.\ne.g. `import { NzButtonModule } from 'ng-zorro-antd/button';`");
    }
    /** @nocollapse */
    NgZorroAntdModule.ctorParameters = function () { return []; };
NgZorroAntdModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgZorroAntdModule });
NgZorroAntdModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgZorroAntdModule_Factory(t) { return new (t || NgZorroAntdModule)(); }, imports: [NzAffixModule,
        NzAlertModule,
        NzAnchorModule,
        NzAutocompleteModule,
        NzAvatarModule,
        NzBackTopModule,
        NzBadgeModule,
        NzButtonModule,
        NzBreadCrumbModule,
        NzCalendarModule,
        NzCardModule,
        NzCarouselModule,
        NzCascaderModule,
        NzCheckboxModule,
        NzCollapseModule,
        NzCommentModule,
        NzDatePickerModule,
        NzDescriptionsModule,
        NzDividerModule,
        NzDrawerModule,
        NzDropDownModule,
        NzEmptyModule,
        NzFormModule,
        NzGridModule,
        NzI18nModule,
        NzIconModule,
        NzInputModule,
        NzInputNumberModule,
        NzLayoutModule,
        NzListModule,
        NzMentionModule,
        NzMenuModule,
        NzMessageModule,
        NzModalModule,
        NzNoAnimationModule,
        NzNotificationModule,
        NzPageHeaderModule,
        NzPaginationModule,
        NzPopconfirmModule,
        NzPopoverModule,
        NzProgressModule,
        NzRadioModule,
        NzRateModule,
        NzResultModule,
        NzSelectModule,
        NzSkeletonModule,
        NzSliderModule,
        NzSpinModule,
        NzStatisticModule,
        NzStepsModule,
        NzSwitchModule,
        NzTableModule,
        NzTabsModule,
        NzTagModule,
        NzTimePickerModule,
        NzTimelineModule,
        NzToolTipModule,
        NzTransButtonModule,
        NzTransferModule,
        NzTreeModule,
        NzTreeSelectModule,
        NzTypographyModule,
        NzUploadModule,
        NzWaveModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgZorroAntdModule, { exports: function () { return [NzAffixModule,
        NzAlertModule,
        NzAnchorModule,
        NzAutocompleteModule,
        NzAvatarModule,
        NzBackTopModule,
        NzBadgeModule,
        NzButtonModule,
        NzBreadCrumbModule,
        NzCalendarModule,
        NzCardModule,
        NzCarouselModule,
        NzCascaderModule,
        NzCheckboxModule,
        NzCollapseModule,
        NzCommentModule,
        NzDatePickerModule,
        NzDescriptionsModule,
        NzDividerModule,
        NzDrawerModule,
        NzDropDownModule,
        NzEmptyModule,
        NzFormModule,
        NzGridModule,
        NzI18nModule,
        NzIconModule,
        NzInputModule,
        NzInputNumberModule,
        NzLayoutModule,
        NzListModule,
        NzMentionModule,
        NzMenuModule,
        NzMessageModule,
        NzModalModule,
        NzNoAnimationModule,
        NzNotificationModule,
        NzPageHeaderModule,
        NzPaginationModule,
        NzPopconfirmModule,
        NzPopoverModule,
        NzProgressModule,
        NzRadioModule,
        NzRateModule,
        NzResultModule,
        NzSelectModule,
        NzSkeletonModule,
        NzSliderModule,
        NzSpinModule,
        NzStatisticModule,
        NzStepsModule,
        NzSwitchModule,
        NzTableModule,
        NzTabsModule,
        NzTagModule,
        NzTimePickerModule,
        NzTimelineModule,
        NzToolTipModule,
        NzTransButtonModule,
        NzTransferModule,
        NzTreeModule,
        NzTreeSelectModule,
        NzTypographyModule,
        NzUploadModule,
        NzWaveModule]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgZorroAntdModule, [{
        type: NgModule,
        args: [{
                exports: [
                    NzAffixModule,
                    NzAlertModule,
                    NzAnchorModule,
                    NzAutocompleteModule,
                    NzAvatarModule,
                    NzBackTopModule,
                    NzBadgeModule,
                    NzButtonModule,
                    NzBreadCrumbModule,
                    NzCalendarModule,
                    NzCardModule,
                    NzCarouselModule,
                    NzCascaderModule,
                    NzCheckboxModule,
                    NzCollapseModule,
                    NzCommentModule,
                    NzDatePickerModule,
                    NzDescriptionsModule,
                    NzDividerModule,
                    NzDrawerModule,
                    NzDropDownModule,
                    NzEmptyModule,
                    NzFormModule,
                    NzGridModule,
                    NzI18nModule,
                    NzIconModule,
                    NzInputModule,
                    NzInputNumberModule,
                    NzLayoutModule,
                    NzListModule,
                    NzMentionModule,
                    NzMenuModule,
                    NzMessageModule,
                    NzModalModule,
                    NzNoAnimationModule,
                    NzNotificationModule,
                    NzPageHeaderModule,
                    NzPaginationModule,
                    NzPopconfirmModule,
                    NzPopoverModule,
                    NzProgressModule,
                    NzRadioModule,
                    NzRateModule,
                    NzResultModule,
                    NzSelectModule,
                    NzSkeletonModule,
                    NzSliderModule,
                    NzSpinModule,
                    NzStatisticModule,
                    NzStepsModule,
                    NzSwitchModule,
                    NzTableModule,
                    NzTabsModule,
                    NzTagModule,
                    NzTimePickerModule,
                    NzTimelineModule,
                    NzToolTipModule,
                    NzTransButtonModule,
                    NzTransferModule,
                    NzTreeModule,
                    NzTreeSelectModule,
                    NzTypographyModule,
                    NzUploadModule,
                    NzWaveModule
                ]
            }]
    }], function () { return []; }, null); })();
    return NgZorroAntdModule;
}());
export { NgZorroAntdModule };
